### PR TITLE
[quant][pt2] Support quantized conv bias in QAT fusion

### PR DESCRIPTION
**Pull request by andrewor14**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #113006
* #112612
* __->__ #112528

Summary: Previously QAT fusion assumes bias is not quantized.
This works for the existing XNNPACKQuantizer, but not for custom
quantizers that wish to quantize the bias. This commit supports
this by adding the necessary patterns. This requires refactoring
the code, however, since it previously assumed that there will
only be one pair of q-dq (from conv weight) in the matched
pattern, and this is no longer true.

Test Plan:
python test/test_quantization.py TestQuantizePT2EQAT.test_qat_conv_bn_bias_derived_qspec

Reviewers: jerryzh168, kimishpatel

Subscribers: jerryzh168, kimishpatel, supriyar

Differential Revision: [D50856377](https://our.internmc.facebook.com/intern/diff/D50856377)

### CODE REVIEW

**Comment by andrewor14** [comment]
@andrewor14 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D50856377).

**Comment by andrewor14** [comment]
@andrewor14 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D50856377).

File: torch/ao/quantization/pt2e/qat_utils.py [hunk 1]
```diff
@@ -771,63 +822,27 @@ def _fold_conv_bn_qat_helper(m: GraphModule, is_cuda: bool) -> GraphModule:
...
-        (_, bn_node) = node_map["bn"]
+        node_map = _get_conv_bn_pattern_nodes(r)
 
         # Step (2): Copy over metadata from original subgraph
         for original_node, replacement_node in node_map.values():
             replacement_node.meta = original_node.meta
 
-        # Step (3): Fold BN weights into conv
+        # Step (3): Copy over args for weight (and optionally bias) q - dq nodes
+        _copy_over_q_dq_args(*node_map["conv_weight_q"])
```
Comment (jerryzh168) [inline torch/ao/quantization/pt2e/qat_utils.py:832]: I'm wondering if these can be simplified if we use these? https://github.com/pytorch/pytorch/blob/main/torch/ao/quantization/pt2e/utils.py#L291

**Review by jerryzh168 — APPROVED** [review]

File: torch/ao/quantization/pt2e/qat_utils.py (hunk 1 above)
Comment (andrewor14) [reply torch/ao/quantization/pt2e/qat_utils.py:832]: Yeah, we can explore that as a BE task along with the conv arg copying. I prefer to do that separately

**Comment by andrewor14** [comment]
@andrewor14 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D50856377).

**Comment by andrewor14** [comment]
@andrewor14 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D50856377).

**Pull request closed without merging**